### DISCUSSION TITLE
Extract the font features in one method?

### DISCUSSION THREAD

**Discussion by JaavierR** [fonttools/fonttools#2873] [Q&A] [answered]
Hi! 

Recently for a project I started working with font files and I found this repo that help me to achieve most of the tasks. But one of them is extract all the font features in a list, and I'm wondering if is exist a direct method to achieve this, for now I'm using a variation of this [script](https://github.com/fonttools/fonttools/blob/8ab6af03c89726cf80ca3c4b755ae1bd0038b5da/Snippets/layout-features.py), that is the most close approach that I found after looking on the docs and the issues/discussions.

Thanks in advance for your help!

**Comment by jsbien** [comment]
I am also interested in the problem and I was unaware of the script. What have you changed in it?

**Comment by ftCLI** [comment, marked as answer]
This is the function I use to get a list feature tags:

```
def get_font_feature_tags(font: TTFont) -> list:

    feature_tags = []

    for table in ("GSUB", "GPOS"):
        if table in font.keys():
            try:
                feature_tags += [tag.FeatureTag for tag in font[table].table.FeatureList.FeatureRecord]
            except KeyError:
                pass

    return sorted(list(set(feature_tags)))
```

**Reply by justvanrossum** [reply]
That's good. Small improvements suggested:
```python
def get_font_feature_tags(font: TTFont) -> list:
    feature_tags = set()
    for table_tag in ("GSUB", "GPOS"):
        if table_tag in font:
            feature_tags.update(
                feature_record.FeatureTag
                for feature_record in font[table_tag].table.FeatureList.FeatureRecord
            )
    return sorted(feature_tags)
```

**Reply by JaavierR** [reply]
Thanks for the answers and suggestion!, I'm currently doing something like that, and It's seems that the lib by itself doesn't have a direct method, so I'm sticking with your solution.

Also thanks @justvanrossum for the suggestion!

@jsbien It was something similar to the solution suggested above, but more messy because I was testing until I ask here.

**Reply by jsbien** [reply]
What is exactly your goal? Are you aware of the human readable TTX format (produced by the ttx program)?
